From 9b6308de80eeecd20c7c88715abd380e31afe77b Mon Sep 17 00:00:00 2001
From: Peng Ma <peng.ma@nxp.com>
Date: Fri, 30 Nov 2018 09:53:12 +0800
Subject: [PATCH 368/706] arm64: dts: ls1028: add qdma node support.

ls1028a different other Layerscape soc, it is a little endian

Signed-off-by: Peng Ma <peng.ma@nxp.com>
(cherry picked from commit ca5e8273988142080830ff5bfe12504a188a7336)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 20 +++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 33e43ed19d32..20f417613e7e 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -709,6 +709,26 @@
 				 <&clockgen 4 1>;
 		};
 
+		qdma: qdma@8380000 {
+			compatible = "fsl,ls1021a-qdma", "fsl,ls1028a-qdma";
+			reg = <0x0 0x8380000 0x0 0x1000>, /* Controller regs */
+			      <0x0 0x8390000 0x0 0x10000>, /* Status regs */
+			      <0x0 0x83a0000 0x0 0x40000>; /* Block regs */
+			interrupts = <0 43 0x4>,
+				     <0 251 0x4>,
+				     <0 252 0x4>,
+				     <0 253 0x4>,
+				     <0 254 0x4>;
+			interrupt-names = "qdma-error", "qdma-queue0",
+				"qdma-queue1", "qdma-queue2", "qdma-queue3";
+			channels = <8>;
+			block-number = <1>;
+			block-offset = <0x10000>;
+			queues = <2>;
+			status-sizes = <64>;
+			queue-sizes = <64 64>;
+		};
+
 		sai1: sai@f100000 {
 			#sound-dai-cells = <0>;
 			compatible = "fsl,vf610-sai";
-- 
2.17.1

